src/app/forms/validators/custom-async-validators.spec.ts

```diff
@@ -0,0 +1,90 @@
+import { MockStore, provideMockStore } from '@ngrx/store/testing';
+import { initialAppState } from '@store/.';
+import { testResultInEdit, TestResultsState } from '@store/test-records';
+import { Observable, Subject, takeUntil } from 'rxjs';
+import { CustomAsyncValidators } from './custom-async-validators';
+
+describe('resultDependantOnCustomDefects', () => {
+  let form: FormGroup;
+  let store: MockStore<TestResultsState>;;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [provideMockStore({ initialState: initialAppState })]
+    });
+
+    store = TestBed.inject(MockStore);
+
+    form = new FormGroup({
+      testResult: new CustomFormControl({ name: 'testResult', type: FormNodeTypes.CONTROL, children: [] }, null)
+    });
+  });
+
+  it('should fail validation when value is "pass" and defects are present', async () => {
+    form.controls['testResult'].patchValue('pass');
+
+    store.overrideSelector(testResultInEdit, mockTestResult());
+
+    const result = await firstEmitted(
+      CustomAsyncValidators.resultDependantOnCustomDefects(store)(form.controls['testResult']) as Observable<ValidationErrors | null>
+    );
+
+    expect(result).toEqual({ invalidTestResult: { message: 'Cannot pass test when defects are present' } });
+  });
+
+  it('should fail validation when value is "fail" but no defects are present', async () => {
+    form.controls['testResult'].patchValue('fail');
+
+    const testResult = mockTestResult();
+    testResult.testTypes = [];
+
+    store.overrideSelector(testResultInEdit, testResult);
+
+    const result = await firstEmitted(
+      CustomAsyncValidators.resultDependantOnCustomDefects(store)(form.controls['testResult']) as Observable<ValidationErrors | null>
+    );
+
+    expect(result).toEqual({ invalidTestResult: { message: 'Cannot fail test when no defects are present' } });
+  });
+
+  it('should fail validation when value is "prs" but no defects are present', async () => {
+    form.controls['testResult'].patchValue('prs');
+
+    const testResult = mockTestResult();
+    testResult.testTypes = [];
+
+    store.overrideSelector(testResultInEdit, testResult);
+
+    const result = await firstEmitted(
+      CustomAsyncValidators.resultDependantOnCustomDefects(store)(form.controls['testResult']) as Observable<ValidationErrors | null>
+    );
+
+    expect(result).toEqual({ invalidTestResult: { message: 'Cannot mark test as PRS when no defects are present' } });
+  });
+
+  it('should pass validation when value is "abandoned"', async () => {
+    form.controls['testResult'].patchValue('abandoned');
+
+    store.overrideSelector(testResultInEdit, mockTestResult());
+
+    const result = await firstEmitted(
+      CustomAsyncValidators.resultDependantOnCustomDefects(store)(form.controls['testResult']) as Observable<ValidationErrors | null>
+    );
+
+    expect(result).toBeNull();
+  });
+});
+
+function firstEmitted<T>(obs$: Observable<T>): Promise<T> {
```

Comment:
Can you not use the `firstValueFrom` operator ?

Reply:
This is great - suggestion applied.